File: services/backend/src/agent_cloud_backend/models/skill.py
import uuid

from sqlalchemy import Boolean, ForeignKey, UniqueConstraint
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column

from agent_cloud_backend.models.base import Base, TimestampMixin, uuid_pk


class Skill(Base, TimestampMixin):
    __tablename__ = "skills"

    id: Mapped[uuid.UUID] = uuid_pk()
    user_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False
    )
    name: Mapped[str] = mapped_column(nullable=False)
    description: Mapped[str] = mapped_column(nullable=False)
    source: Mapped[str] = mapped_column(nullable=False)  # registry | uploaded
    version: Mapped[str] = mapped_column(nullable=False, default="0.0.0")
    requires: Mapped[dict] = mapped_column(JSONB, default=dict, nullable=False)
    package_ref: Mapped[str] = mapped_column(nullable=False)

    __table_args__ = (UniqueConstraint("user_id", "name", name="uq_skill_user_name"),)


class AgentSkillEnable(Base):
    __tablename__ = "agent_skill_enables"

    agent_config_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("agent_configs.id", ondelete="CASCADE"), primary_key=True
    )
    skill_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("skills.id", ondelete="CASCADE"), primary_key=True
    )
    enabled: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

File: apps/mobile/lib/models/skill.dart
/// 已安装技能。
class Skill {
  final String id;
  final String name;
  final String description;
  final String source; // builtin / workspace / upload 等
  final String version;

  const Skill({
    required this.id,
    required this.name,
    required this.description,
    required this.source,
    required this.version,
  });

  factory Skill.fromJson(Map<String, dynamic> j) => Skill(
        id: j['id'] as String,
        name: j['name'] as String,
        description: j['description'] as String? ?? '',
        source: j['source'] as String? ?? '',
        version: j['version'] as String? ?? '',
      );
}
